doc/man7/ossl-guide-quic-server-block.pod

```diff
@@ -0,0 +1,260 @@
+     * SSL_CTX_set_default_verify_paths() is needed.  The server's own
+     * certificate chain is assumed valid.
+     */
+    SSL_CTX_set_verify(ctx, SSL_VERIFY_NONE, NULL);
+
+
+QUIC also dictates using Application-Layer Protocol Negotiation [ALPN] to select
+an application protocol.  We use L<SSL_CTX_set_alpn_select_cb(3)> for this
+purpose.  We can pass a callback which will be called for each connection to
+select an ALPN the server considers acceptable.  In this case, we only accept
+"http/1.0" and "hq-interop".
+
+    /* Setup ALPN negotiation callback to decide which ALPN is accepted. */
+    SSL_CTX_set_alpn_select_cb(ctx, select_alpn, NULL);
+
+That is all the setup that we need to do for the B<SSL_CTX>.  Next, we create a
+UDP socket and bind to it on localhost.
+
+    /* Retrieve the file descriptor for a new UDP socket */
+    if ((fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)) < 0) {
+        fprintf(stderr, "cannot create socket");
+        goto err;
+    }
+
+    sa.sin_family = AF_INET;
+    sa.sin_port = htons(port);
+
+    /* Bind to the new UDP socket on localhost */
+    if (bind(fd, (const struct sockaddr *)&sa, sizeof(sa)) < 0) {
+        fprintf(stderr, "cannot bind to %u\n", port);
+        BIO_closesocket(fd);
+        goto err;
+    }
+
+To run the QUIC server, we create a B<SSL_LISTENER> to listen for incoming
+connections.  We provide it with the bound UDP port to then explicitly begin
+listening for new connections.
+
+    /*
+     * Create a new QUIC listener. Listeners, and other QUIC objects, default
+     * to operating in blocking mode. The configured behaviour is inherited by
+     * child objects.
+     */
+    if ((listener = SSL_new_listener(ctx, 0)) == NULL) {
+        goto err;
+    }
+
+    /* Provide the listener with our UDP socket. */
+    if (!SSL_set_fd(listener, fd))
+        goto err;
+
+    /* Begin listening. */
+    if (!SSL_listen(listener))
+        goto err;
+
+=head2 Server loop
+
+The server now enters a "forever" loop, handling one client connection at a
+time.  Before each connection, we clear the OpenSSL error stack so that any
+error reports are related to just the new connection.
+
+    /* Pristine error stack for each new connection */
+    ERR_clear_error();
+
+At this point, the server blocks to accept the next client.
+L<SSL_accept_connection(3)> will return an accepted connection within a fresh
+SSL, in which the handshake will have already occurred.
+
+    /* Block while waiting for a client connection */
+    conn = SSL_accept_connection(listener, 0);
+    if (conn == NULL) {
+        fprintf(stderr, "error while accepting connection\n");
+        goto err;
+    }
+
+With the handshake complete, the server echoes client input back to the client
+in a loop.
+
+    while (SSL_read_ex(conn, buf, sizeof(buf), &nread) > 0) {
+        if (SSL_write_ex(conn, buf, nread, &nwritten) > 0 &&
+            nwritten == nread) {
+            break;
```

Comment:
Hmmm. I just noticed this. This doesn't look quite right to me. In the `tls-server-block.c` equivalent we have a `continue` here - which is what I would have expected. A break means we could prematurely stop echoing client data before it has finished sending us stuff.